.travis_update_ghpages.sh

```diff
@@ -1,25 +1,43 @@
 #!/bin/bash
+
+set -o errexit -o nounset
+
+YELLOW=$'\e[1;33m'
+GREEN=$'\e[1;32m'
+RED=$'\e[1;31m'
+RESET=$'\e[0m'
+
 # Only publish from the main repository's master branch
 REPO_NAME="HydraCG/Specifications"
-if [ "$TRAVIS_REPO_SLUG" != "$REPO_NAME" ] || [ "$TRAVIS_BRANCH" != "master" ] || [ "$TRAVIS_PULL_REQUEST" != "false" ]; then exit; fi
-echo -e "Generating gh-pages...\n"
+if [ "$TRAVIS_REPO_SLUG" != "$REPO_NAME" ] || [ "$TRAVIS_BRANCH" != "master" ] || [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+  echo -e "${RED}This commit was made against $TRAVIS_BRANCH and not the master. Don't update gh-pages.${RESET}"
+  exit;
+fi
+
+echo -e "${YELLOW}Updating gh-pages...${RESET}"
+
+echo -e "${YELLOW}Generated files in $PWD/drafts/diagram${RESET}"
+ls drafts/diagram
 
 # Checkout the gh-pages branch
+echo -e "${YELLOW}Cloning and updating GitHub repository${RESET}"
 REPO_PATH=$PWD
 pushd $HOME
-git clone --quiet --branch=gh-pages https://${GH_TOKEN}@github.com/$REPO_NAME gh-pages 2>&1 > /dev/null
+git clone --branch=gh-pages https://${GH_TOKEN}@github.com/$REPO_NAME gh-pages
 cd gh-pages
-
-# Don't update if already at the latest version
-if [[ `git log -1 --pretty=%B` == *$TRAVIS_COMMIT* ]]; then exit; fi
+mkdir -p drafts/diagram/
 
 # Update documents
-cp -r $REPO_PATH .
+cp -rf $REPO_PATH/drafts/diagram/* drafts/diagram/
 
 # Commit and push latest version
-git add .
 git config user.name  "Travis"
-git config user.email "travis@travis-ci.org"
+git config user.email "travis@hydra-cg.com"
+git add -A .
+git status
 git commit -m "Update to $TRAVIS_COMMIT."
-git push -fq origin gh-pages 2>&1 > /dev/null
+git push -q origin gh-pages
```

Comment:
Unsure about all the errors that could happen here, but basically `origin` uses the secret `GH_TOKEN` variable (which I have set in Travis). Perhaps if there is a connection drop or whatever, the reported error could be "could not push to SECRETTOKEN@github.com" and then all my repositories are screwed ;-)